src/ol/PluggableMap.js

```diff
@@ -1162,12 +1137,36 @@ class PluggableMap extends BaseObject {
         removeEventListener(EventType.RESIZE, this.handleResize_, false);
         this.handleResize_ = undefined;
       }
+      this.mapBrowserEventHandler_.dispose();
+      this.viewport_.removeEventListener(
+        EventType.CONTEXTMENU,
+        this.boundHandleBrowserEvent_
+      );
+      this.viewport_.removeEventListener(
+        EventType.WHEEL,
+        this.boundHandleBrowserEvent_
+      );
+      if (this.handleResize_ !== undefined) {
+        removeEventListener(EventType.RESIZE, this.handleResize_, false);
+        this.handleResize_ = undefined;
+      }
     } else {
       targetElement.appendChild(this.viewport_);
       if (!this.renderer_) {
         this.renderer_ = this.createRenderer();
       }
 
+      const handleBrowserEvent = this.handleBrowserEvent.bind(this);
```

Comment:
Curious as to why we bind again instead of using `boundHandleBrowserEvent_`. Will a different bind be able to get matched over `addEventListener` and `removeEventListener`?

Reply:
I just realized I did not push everything I had locally. Hold on, update to come.